### PR TITLE
Remove most calls to time.Now

### PR DESCRIPTION
**Pull request by mrd0ll4r**
Before merging, I want to check what this does to our metrics.
Still, gives me a few hundred more reqs/s on a linux machine, and here are some storage benchmarks:

```
benchmark                                 old ns/op     new ns/op     delta
BenchmarkPut-2                            410           361           -11.95%
BenchmarkPut-4                            487           455           -6.57%
BenchmarkPut1k-2                          517           394           -23.79%
BenchmarkPut1k-4                          521           486           -6.72%
BenchmarkPut1kInfohash-2                  276           266           -3.62%
BenchmarkPut1kInfohash-4                  138           134           -2.90%
BenchmarkPut1kInfohash1k-2                279           267           -4.30%
BenchmarkPut1kInfohash1k-4                144           136           -5.56%
BenchmarkPutDelete-2                      1227          1223          -0.33%
BenchmarkPutDelete-4                      3089          2506          -18.87%
BenchmarkPutDelete1k-2                    1236          1361          +10.11%
BenchmarkPutDelete1k-4                    2850          2827          -0.81%
BenchmarkPutDelete1kInfohash-2            1241          1266          +2.01%
BenchmarkPutDelete1kInfohash-4            2596          2591          -0.19%
BenchmarkPutDelete1kInfohash1k-2          1249          1436          +14.97%
BenchmarkPutDelete1kInfohash1k-4          2636          2193          -16.81%
BenchmarkDeleteNonexist-2                 216           168           -22.22%
BenchmarkDeleteNonexist-4                 192           191           -0.52%
BenchmarkDeleteNonexist1k-2               155           170           +9.68%
BenchmarkDeleteNonexist1k-4               190           192           +1.05%
BenchmarkDeleteNonexist1kInfohash-2       142           138           -2.82%
BenchmarkDeleteNonexist1kInfohash-4       78.4          79.5          +1.40%
BenchmarkDeleteNonexist1kInfohash1k-2     145           139           -4.14%
BenchmarkDeleteNonexist1kInfohash1k-4     78.4          82.9          +5.74%
BenchmarkPutGradDelete-2                  2006          1821          -9.22%
BenchmarkPutGradDelete-4                  3558          3715          +4.41%
BenchmarkPutGradDelete1k-2                1757          1890          +7.57%
BenchmarkPutGradDelete1k-4                4160          3871          -6.95%
BenchmarkPutGradDelete1kInfohash-2        1830          1786          -2.40%
BenchmarkPutGradDelete1kInfohash-4        3554          3123          -12.13%
BenchmarkPutGradDelete1kInfohash1k-2      1862          1796          -3.54%
BenchmarkPutGradDelete1kInfohash1k-4      3332          3743          +12.33%
BenchmarkGradNonexist-2                   515           407           -20.97%
BenchmarkGradNonexist-4                   513           481           -6.24%
BenchmarkGradNonexist1k-2                 547           454           -17.00%
BenchmarkGradNonexist1k-4                 557           523           -6.10%
BenchmarkGradNonexist1kInfohash-2         283           276           -2.47%
BenchmarkGradNonexist1kInfohash-4         157           138           -12.10%
BenchmarkGradNonexist1kInfohash1k-2       285           276           -3.16%
BenchmarkGradNonexist1kInfohash1k-4       156           155           -0.64%
BenchmarkAnnounceLeecher-2                9623          11358         +18.03%
BenchmarkAnnounceLeecher-4                6215          6126          -1.43%
BenchmarkAnnounceLeecher1kInfohash-2      11709         11742         +0.28%
BenchmarkAnnounceLeecher1kInfohash-4      7435          7320          -1.55%
BenchmarkAnnounceSeeder-2                 15102         9438          -37.50%
BenchmarkAnnounceSeeder-4                 5313          5837          +9.86%
BenchmarkAnnounceSeeder1kInfohash-2       11720         11832         +0.96%
BenchmarkAnnounceSeeder1kInfohash-4       7357          7737          +5.17%
```

Most of the extremes are probably not reproducible. Overall decrease is nice though. Not reproducible on single-core because of the additional goroutine, unfortunately :(

### CODE REVIEW

File: frontend/udp/frontend.go [hunk 1]
```diff
@@ -201,13 +203,20 @@ func (t *Frontend) listenAndServe() error {
 			}
 
 			// Handle the request.
-			start := time.Now()
+			start := time.Time{}
```
Comment (jzelinskie) [inline frontend/udp/frontend.go]: var start time.Time

File: storage/memory/peer_store.go [hunk 2]
```diff
@@ -96,6 +97,25 @@ func New(cfg Config) (storage.PeerStore, error) {
...
+			select {
+			case <-ps.closing:
+				t.Stop()
+				select {
+				case <-t.C:
+				default:
+				}
+				return
+			case now := <-t.C:
+				ps.setClock(now.UnixNano())
```
Comment (jzelinskie) [inline storage/memory/peer_store.go]: do we really need nano?

Comment (mrd0ll4r) [inline storage/memory/peer_store.go]: `UnixNano` doesn't cause another syscall. `t.C` already gives us the current time, so we might as well use it :)

**Comment by mrd0ll4r** [comment]
Stats still work even without timing them :)

**Pull request merged**